src/commands/Birthday/set-birthday.ts

```diff
@@ -45,38 +45,35 @@ export class UserCommand extends SkyraCommand {
 		};
 	}
 
-	private async handleArguments(args: SkyraCommand.Args): Promise<DateWithOptionalYear> {
-		const result = await args.pickResult(UserCommand.dateWithOptionalYear);
-		if (result.success) return result.value;
-
-		const birthDate = await args.rest('date');
-
-		// The world's oldest human alive was born in January 1903:
-		if (birthDate.getTime() > Date.now() || birthDate.getFullYear() < 1903) this.error(LanguageKeys.Commands.Misc.SetBirthdayInvalidDate);
-		return { year: birthDate.getUTCFullYear(), month: birthDate.getUTCMonth() + 1, day: birthDate.getUTCDate() };
-	}
-
 	private isCorrectlyConfigured(birthdayRole: string | Nullish, birthdayChannel: string | Nullish, birthdayMessage: string | Nullish) {
 		// A birthday role, or a channel and message must be configured:
 		return !isNullish(birthdayRole) || (!isNullish(birthdayChannel) && !isNullish(birthdayMessage));
 	}
 
-	private static readonly dateRegExp = /^(?:(\d{4})[-/])?(\d{1,2})[-/](\d{1,2})/;
-	private static dateWithOptionalYear = Args.make<DateWithOptionalYear>((parameter, { argument }) => {
-		const result = this.dateRegExp.exec(parameter);
-		if (result === null) return Args.error({ argument, parameter, identifier: LanguageKeys.Commands.Misc.SetBirthdayInvalidDate });
+	private static readonly dateWithOptionalYear = Args.make<DateWithOptionalYear>((parameter, { argument, args }) => {
+		const format = args.t(LanguageKeys.Globals.DateFormat);
+		const regExp = getDateFormat(format, args.t.lng);
+		const result = regExp.exec(parameter);
+		if (result === null) {
+			return Args.error({
+				argument,
+				parameter,
+				identifier: LanguageKeys.Commands.Misc.SetBirthdayInvalidDate,
+				context: { formatWithYear: format, formatWithoutYear: removeYear(format) }
+			});
+		}
 
-		const year = result[1] === undefined ? null : Number(result[1]);
+		const year = result.groups!.year === undefined ? null : Number(result.groups!.year);
```

Comment:
```suggestion
		const year = result.groups?.year === undefined ? null : Number(result.groups.year);
```

Not sure if that second ! can be removed when making the first a ? but I think so

Reply:
`groups` being `Record<string, string> | undefined` is being so because it's generic for all RegExps, however, the RegExps we're using all use groups, and therefore, this can never be undefined.